examples/encoding/plot_encoding_model.py

```diff
@@ -0,0 +1,145 @@
+stim_events = binarizer.fit_transform(events[:, 0], event_ids=events[:, -1])
+sig_event = modulate_noise(stim_events[0], sfreq, delays_sig,
+                           weights=weights, snr=snr)
+info = mne.create_info(['ch'], sfreq, ch_types=['eeg'])
+sig_cont = mne.io.RawArray(sig_cont[np.newaxis, :], info)
+sig_event = mne.io.RawArray(sig_event[np.newaxis, :], info)
+
+
+# -- Preparing model features --
+# Iterate through our events-based and continuous datasets
+data_iterator = [(stim_events, sig_event, 'Events'),
+                 (stim_cont[np.newaxis, :], sig_cont, 'Continuous')]
+
+# Define a training / test set of indices
+ixs = np.arange(sig_cont.n_times)
+ix_split = int(.8 * ixs.shape[0])
+tr = ixs[:ix_split]
+tt = ixs[ix_split:]
+
+
+# -- Define preprocessing pipelines for X/y --
+# For creating time lags
+delays_model = delays_sig
+delayer = DataDelayer(delays=delays_model, sfreq=sig_cont.info['sfreq'])
+
+# To subset training data
+sub_tr = DataSubsetter(tr)
+sub_tt = DataSubsetter(tt)
+
+# Now put them together into pipelines
+pipe_x = Pipeline([('delays', delayer), ('subset', sub_tr)])
+pipe_y = Pipeline([('subset', sub_tr)])
+preproc_x_pred = Pipeline([('delays', delayer), ('subset', sub_tt)])
+
+
+# -- Fit the model, iterating through alphas --
+alphas = np.logspace(0, 2, 4)
+f, axs_coef = plt.subplots(2, len(alphas), figsize=(4 * len(alphas), 8),
+                           sharex=True, sharey=True)
+f, axs_pred = plt.subplots(2, len(alphas), figsize=(4 * len(alphas), 8),
+                           sharex=True, sharey=False)
+for alpha, axcol_coef, axcol_pred in zip(alphas, axs_coef.T, axs_pred.T):
+    axcol_coef[0].set_title('Alpha: %s' % np.round(np.log10(alpha), 2))
+    axcol_pred[0].set_title('Alpha: %s' % np.round(np.log10(alpha), 2))
+
+    # Define our estimator and create the model
+    clf = Ridge(alpha=alpha)
+    mod = EncodingModel(est=clf, preproc_x=pipe_x, preproc_y=pipe_y)
+
+    # Fit the model and plot coefficients
+    iter_fit = zip(data_iterator, axcol_coef, axcol_pred)
+    for (X, y, title), ax_coef, ax_pred in iter_fit:
+        mod.fit(X, y._data)
+
+        # Plot coefficients
+        ax = ax_coef
+        ax.plot(delays_model, mod.coef_.squeeze())
+        ax.plot(delays_sig, weights)
+        ax.set_xlabel('Lag (s)')
+
+        # Plot predictions (on same data, so kinda cheating)
+        predicted = mod.predict(X, preproc_x=preproc_x_pred)
+        ax_pred.plot(y._data[:, tt].squeeze(), color='k', alpha=.3)
+        ax_pred.plot(predicted.squeeze(), color='r')
+
+# Cleaning up plots
+axs_coef[0, -1].legend(['Model coefficients', 'True weights'],
+                       loc=(.95, .9), fontsize='small')
+axs_pred[0, -1].legend(['Actual signal', 'Model prediction'],
+                       loc=(.95, .9), fontsize='small')
+for axgrp in [axs_coef, axs_pred]:
+    axgrp[0, 0].set_ylabel('Continuous input')
+    axgrp[1, 0].set_ylabel('Events-based input (rERP)')
+plt.show()
```

Comment:
Do you have this already lying around as a gist perhaps? (Yes I am that lazy ...)